---
title: "PreprocessQC"
author: "Victor, Ming, Michael"
output: 
  html_document: 
    keep_md: yes
---

We are following this Minfi tutorial:
https://www.bioconductor.org/help/course-materials/2015/BioC2015/methylation450k.html

And we used some additional Minfi functions for QC following the advice of helpful grad students (Thanks Robinson lab!) and Dr. Meaghan's STAT540 DNA methylation lecture <add lecture link>

For general information on Minfi, consult the user guide:
http://bioconductor.org/packages/release/bioc/vignettes/minfi/inst/doc/minfi.pdf

Further info about the functions and capabilities of Minfi can be found in the Minfi reference manual:
http://bioconductor.org/packages/release/bioc/manuals/minfi/man/minfi.pdf

For the relevant paper with more comprehensive discussion of Minfi:
https://academic.oup.com/bioinformatics/article/30/10/1363/267584/Minfi-a-flexible-and-comprehensive-Bioconductor



```{r, load libraries, message=FALSE}
library(minfi)
library(dplyr)
library(ggplot2)
```
#1.0 Introduction

Minfi has a couple of different classes because...

The following classes and their details:

RGChannelSet - Raw intensities 
MethylSet - 
GenomicMethylSet
RatioSet
GenomicRatioSet

different classes allow for ' a flexible framework for method development and analyses'
The natural starting point for analysis is usually the GenomicRatioSet class.

#1.1 Load data
```{r, read data}
<<<<<<< HEAD
getwd()
#input the right base directory
=======
getwd() #input the right base directory
>>>>>>> 2baff207ff6dfb73c150e36872bed7955f5e202f
setwd("./data/Raw Data/")
basedir <- getwd()
samplesheet <- read.metharray.sheet(basedir, recursive = TRUE) # read in sample sheet .csv file
Eth_rgset <- read.metharray.exp(targets = samplesheet) # read in iDAT files using sample sheet
```


```{r}
pheno <- pData(Eth_rgset) # phenotype data (from sample sheet)
pheno[,1:6]
mani <- getManifest(Eth_rgset) # manifest **What is this? explain
```

#1.2 Create Classes
Generating MethylSet and RatioSet
```{r}
MSet <- preprocessRaw(Eth_rgset) 
RSet <- ratioConvert(MSet, what = "both", keepCN = TRUE)
beta <- getBeta(RSet)
```

Get GenomicRatioSet
```{r}
GRset <- mapToGenome(RSet)# Various functions available to access GRSet data
gr <- granges(GRset)
```

#2. QC
--------------------------------------------------------------------------------------------------------------------------------------------
Note: I think we should move all of the QC to here, so that it is obvious that we do QC before normalization -victor
Note: I'm just going to move your section over here Ming, -Victor
--------------------------------------------------------------------------------------------------------------------------------------------

```{r}
qc <- getQC(MSet)
plotQC(qc)
```
--Talk about above plot here. Good / Bad?

```{r}
densityPlot(MSet, sampGroups = pheno$Ethnicity) #bit hard to see
```
--Talk about above plot

```{r}
densityBeanPlot(MSet, sampGroups = pheno$Ethnicity) #This is pretty messy, can we make it look betteR? (split into two graphs maybe?)
```
--Talka bout plot

```{r}
controlStripPlot(Eth_rgset, controls="BISULFITE CONVERSION II") #Red is type 1, green is type 2
```
-- Talk about above

```{r}
mdsPlot(Eth_rgset, sampNames = pheno$Sample_Name, sampGroups = pheno$Ethnicity)
```

Comparing preprocessing methodology
```{r}
MSet.raw <- preprocessRaw(Eth_rgset)
MSet.raw <- MSet.raw[order(featureNames(MSet.raw)), ]

MSet.fnorm <- preprocessFunnorm(Eth_rgset, nPCs = 2, sex = NULL, bgCorr = TRUE, dyeCorr = TRUE, verbose = TRUE)
MSet.fnorm <- MSet.fnorm[order(featureNames(MSet.fnorm)), ]

MSet.qnorm <- preprocessQuantile(Eth_rgset, fixOutliers = TRUE, removeBadSamples = TRUE, badSampleCutoff = 10.5, quantileNormalize = TRUE, stratified = TRUE, mergeManifest = FALSE, sex = NULL)
MSet.qnorm <- MSet.qnorm[order(featureNames(MSet.qnorm)),]
    
probeTypes <- data.frame(Name = featureNames(MSet.raw),
                         Type = getProbeType(MSet.raw))
par(mfrow = c(2, 2))


#legendpos = "btm" is used to generate an error to remove the legend all together. 
plotBetasByType(MSet.raw[,1], main = "Raw", legendPos = "btm")
plotBetasByType(getBeta(MSet.fnorm[,1]), probeTypes = probeTypes, main = "funNorm_noob", legendPos = "btm")
plotBetasByType(getBeta(MSet.qnorm[,1]), probeTypes = probeTypes, main = "Quantile", legendPos = "btm")
```

3. Normalization
Try different preporcessing methods:

* Noob: "Implements the noob background subtraction method with dye-bias normalization"
* Functional Normalization: "This function applies the preprocessNoob function as a first step for background substraction, and uses the first two principal components of the control probes to infer the unwanted variation"
* Quantile Normalization: "Implements stratified quantile normalization preprocessing"

```{r}
# noob 
eth_proproc_noob <- preprocessNoob(Eth_rgset)
eth_proproc_noob
# functional normalization
eth_preproc_funnorm <- preprocessFunnorm(Eth_rgset)
eth_preproc_funnorm
# quantile normalization
eth_preproc_quant <- preprocessQuantile(Eth_rgset,fixOutliers = TRUE,
  removeBadSamples = FALSE, quantileNormalize = TRUE, stratified = TRUE, 
  mergeManifest = FALSE, sex = NULL)
eth_preproc_quant
## later on we can use getSex() on preprocessed data to double check genders for each observation
```


Compare different preprocessing methods:

Credit: https://github.com/swils6/PE_IUGR_DataPreprocessing/blob/master/PE_IUGR_FNormalization_Jan2016.md

```{r}

probeTypes <- data.frame(Name = featureNames(eth_proproc_noob),
                         Type = getProbeType(eth_proproc_noob))


par(mfrow = c(1, 2))
# can't plot noob for now, will fix later
#plotBetasByType(eth_preproc_noob, main = "Noob")
plotBetasByType(getBeta(eth_preproc_funnorm[,1]), main = "Functional Normalization", probeTypes = probeTypes)
plotBetasByType(getBeta(eth_preproc_quant[,1]), main = "Quantile Normalization", probeTypes = probeTypes)


```

A good preprocessing method should make the peaks of type 1 & 2 probe distributions close together, so functional normalization appears to be better.

<<<<<<< HEAD
### Approach of preprocessing without FunNorm and Quantile
Generating MethylSet and RatioSet, using preprocessingRaw()
```{r}
MSet <- preprocessRaw(Eth_rgset) 
RSet <- ratioConvert(MSet, what = "both", keepCN = TRUE)
beta <- getBeta(RSet)
```

Get GenomicRatioSet
=======
>>>>>>> 2baff207ff6dfb73c150e36872bed7955f5e202f
```{r}
# sex check
eth_preproc_funnorm@colData$sex
predicted_sex <- eth_preproc_funnorm@colData$predictedSex
for (i in 1:length(predicted_sex)){
  if (predicted_sex[i] == "F") {predicted_sex[i] = "FEMALE"}
  else {predicted_sex[i] = "MALE"} }
predicted_sex == eth_preproc_funnorm@colData$sex
## all match
```

<<<<<<< HEAD
QC
[Resources used](https://www.bioconductor.org/help/course-materials/2015/BioC2015/methylation450k.html#preprocessnoob)
```{r}
qc <- getQC(MSet)
plotQC(qc)
densityPlot(MSet, sampGroups = pheno$Sample_Group)
densityBeanPlot(MSet, sampGroups = pheno$Sample_Group)
controlStripPlot(Eth_rgset, controls="BISULFITE CONVERSION II")
#Red is type 1, green is type 2
```

Comparing preprocessing methodology
```{r}
MSet.raw <- preprocessRaw(Eth_rgset)
MSet.raw <- MSet.raw[order(featureNames(MSet.raw)), ]

MSet.fnorm <- preprocessFunnorm(Eth_rgset, nPCs = 2, sex = NULL, bgCorr = TRUE, dyeCorr = TRUE, verbose = TRUE)
MSet.fnorm <- MSet.fnorm[order(featureNames(MSet.fnorm)), ]

MSet.qnorm <- preprocessQuantile(Eth_rgset, fixOutliers = TRUE, removeBadSamples = TRUE, badSampleCutoff = 10.5, quantileNormalize = TRUE, stratified = TRUE, mergeManifest = FALSE, sex = NULL)
MSet.qnorm <- MSet.qnorm[order(featureNames(MSet.qnorm)),]
    
probeTypes <- data.frame(Name = featureNames(MSet.raw),
                         Type = getProbeType(MSet.raw))
par(mfrow = c(2, 2))


#legendpos = "btm" is used to generate an error to remove the legend all together. Error is expected.
plotBetasByType(MSet.raw[,1], main = "Raw", legendPos = "btm")
plotBetasByType(getBeta(MSet.fnorm[,1]), probeTypes = probeTypes, main = "funNorm_noob", legendPos = "btm")
plotBetasByType(getBeta(MSet.qnorm[,1]), probeTypes = probeTypes, main = "Quantile", legendPos = "btm")
```

> Decided to use FunNorm/ Noob to do precprocessing

```{r}
#Sex check
#Preprocessing annotation of sex due to density of X-chrom should match our phenotype data; QC step.
predictedSex <- getSex(GRset)$predictedSex

```

### Checking for the presence of SNPs inside the probe body or CpG or at the nucleotide extension
```{r}
snps <- getSnpInfo(GRset)
apply(snps, 2, function(x) length(which(!is.na(x))))

# Adding SNP informtion to GenomicRatioSet object (GRset)
GRset <- addSnpInfo(GRset)

#Dropping probes with a SNP in either CpG interrogation site or Single Base Extension site. Why tho?

# 
GRset <- dropLociWithSnps(GRset, snps=c("SBE","CpG"), maf=0)
```



=======
```{r}
# check presence of SNPs inside probe body or single nucleotide extensions
snps <- getSnpInfo(eth_preproc_funnorm)
str(snps@listData$Probe_rs)

eth_preproc_funnorm <- addSnpInfo(eth_preproc_funnorm)
# drop the probes that contain either a SNP at the CpG interrogation or at the single nucleotide extension
eth_preproc_funnorm_rmSNP <- dropLociWithSnps(eth_preproc_funnorm, snps=c("SBE","CpG"), maf=0)
eth_preproc_funnorm_rmSNP
```

4. Export processed data files (data.txt, and des.txt)
>>>>>>> 2baff207ff6dfb73c150e36872bed7955f5e202f
