---
title: "Differential methylation analysis (of Processed Data) with limma"
author: "Nivretta Thatra"
output: github_document
---

#Step 0: Load Packages and Data

###Installed the necessary dependencies by: 

```{r, load libraries, message=FALSE}

#install.packages("gplots")

library(gplots)
library(limma)

```

###Load in pre-processed data from dropbox.

```{r}

#setwd("../Limma")


#download preprocessed data from this dropbox link directly into Limma folder
#data.txt is in gitignore, be sure not to try to push to github
#download.file("https://www.dropbox.com/s/s4xv0k0vsl0ka0r/data.txt?dl=1", "data.txt")

#using dropbox because google drive links do not work

data <- read.table("data.txt")

design <- read.csv("des.txt", sep="\t", header=TRUE)

```

#Step 1: ID differentially methylated sites by Ethnicity only

We can use a linear model to identify differentially methylated probes with `limma`, as shown in this older 540 [seminar](http://www.ugrad.stat.ubc.ca/~stat540/seminars/seminar08_methylation.html).

Create a design matrix looking at just the effect of ethnicity.

```{r design matrix}
des.mat <- model.matrix(~Ethnicity, design)
```

Fit model to obtain top differentially methylated CpG sites.

```{r fit linear model}

DMRfit <- lmFit(data, des.mat)
DMRfitEb <- eBayes(DMRfit)

DMR <- topTable(DMRfitEb, coef = "EthnicityCaucasian", number = Inf, p.value = 0.01, adjust.method = "BH")

knitr::kable(head(DMR))  # top hits 

```

So using a cutoff of FDR = 0.01, we identified 106 CpG sites that are differentially methylated between Caucasian and Asian genetic ancestry. Now we can make some plots to check these hits.

#Step 3: Plotting

```{r heatmap of beta values of top 100 Limma hits}

#get the top 100 differentially methylated CpG sites & values for all samples
DMR100 <- topTable(DMRfitEb, coef = "EthnicityCaucasian", number = 100)
DMR.CpG <- t(as.matrix(subset(data, rownames(data) %in% rownames(DMR100))))
str(DMR.CpG, max.level = 0)

limma_top100_ethinicity <- list(rownames(DMR100))

#write_rds(list(rownames(DMR100)), path = "limma_top100_ethinicity.rds")

#rename the samples before plotting
meta <- cbind(design, DMR.CpG)
rownames(DMR.CpG) <- paste(rownames(DMR.CpG), sub(c(rownames(DMR.CpG)), "", meta$Ethnicity), sep = "_")

#plot heatmap
col <- c(rep("darkgoldenrod1", times = nrow(DMR.CpG)))
col[grepl("Asian", rownames(DMR.CpG))] <- "forestgreen"
op <- par(mai = rep(0.5, 4))
heatmap.2(DMR.CpG, col = redblue(256), RowSideColors = col, density.info = "none", 
    trace = "none", Rowv = TRUE, Colv = TRUE, labCol = FALSE, labRow = FALSE, 
    dendrogram = "row", margins = c(1, 5), main = "Differentially expressed genes 
    by Ethnicity")
legend("topright", c("Caucasian", "Asian"), col = c("darkgoldenrod1", "forestgreen"), 
    pch = 15)
par(op)
```

##Step 4: Check for interaction effects

Create a design matrix looking at the effect of ethnicity when taking into account ethnicity and sex. The Sample_groups (control, lopet, IUGR) are shown in literature to not affect CpG site methylation, so let's just check for gender.

```{r design matrix gender}
des.mat.gender <- model.matrix(~Ethnicity*sex, design)
```

Fit model to obtain top differentially methylated CpG sites.

```{r fit linear model gender}

DMRfit.gender <- lmFit(data, des.mat.gender)
DMRfitEb.gender <- eBayes(DMRfit.gender)

DMR.gender <- topTable(DMRfitEb.gender, coef = "EthnicityCaucasian", number = Inf, p.value = 0.01, adjust.method = "BH")

knitr::kable(head(DMR.gender))  # top hits 

#save results in data folder
#limma_fdr.01_ethnicity.gender <- rownames(DMR.gender)
#write(limma_fdr.01_ethnicity.gender, file = "limma_fdr0.01_ethnicity_accountingforGender.txt")

```

So using a cutoff of FDR = 0.01, we identified just 13 CpG sites that are differentially methylated between Caucasian and Asian genetic ancestry. 

##Step 5: Plot top 100 Limma hits when accounting for Ethnicity and gender interaction effect

```{r heatmap of beta values of top 100 Limma hits with interaction }

#get the top 100 differentially methylated CpG sites & values for all samples
DMR100.gender <- topTable(DMRfitEb.gender, coef = "EthnicityCaucasian", number = 100)
DMR.CpG.gender <- t(as.matrix(subset(data, rownames(data) %in% rownames(DMR100.gender))))

#save results in data folder
#limma_top100_ethnicity.gender <- rownames(DMR100.gender)
#write(limma_top100_ethnicity.gender, file = "limma_top100_ethnicity_accountingforGender.txt")

#rename the samples before plotting
meta2 <- cbind(design, DMR.CpG.gender)
rownames(DMR.CpG.gender) <- paste(rownames(DMR.CpG.gender), sub(c(rownames(DMR.CpG.gender)), "", meta2$Ethnicity), sep = "_")

#plot heatmap
col <- c(rep("darkgoldenrod1", times = nrow(DMR.CpG.gender)))
col[grepl("Asian", rownames(DMR.CpG.gender))] <- "forestgreen"
op <- par(mai = rep(0.5, 4))
heatmap.2(DMR.CpG.gender, col = redblue(256), RowSideColors = col, density.info = "none", 
    trace = "none", Rowv = TRUE, Colv = TRUE, labCol = FALSE, labRow = FALSE, 
    dendrogram = "row", margins = c(1, 5), main = "Differentially expressed genes by Ethnicity, 
    accounting for Gender")
legend("topright", c("Caucasian", "Asian"), col = c("darkgoldenrod1", "forestgreen"), 
    pch = 15)
par(op)
```

##Step 6: Check with GLMnet predictions

Hopefully there is overlap between the CpG sites detected by GLMnet and the ones detected in linear regression analysis. Let's check.

```{r Overlap with GLMnet}

#load in the CpG sites predicted with GLMnet
interestingSites <- read.table("../../Data/Processed Data/predictorsGlmnet.txt")
colnames(interestingSites) <- c("SiteID")

#list of 13 differentially methylated sites
DMR.gender.sites <- data.frame(rownames(DMR.gender))
colnames(DMR.gender.sites) <- c("SiteID")

#find overlaps
overlaps <- merge(interestingSites, DMR.gender.sites)
overlaps

```

There are 5 CpG sites that overlap - which is promising!
